doc/tor.md

```diff
@@ -51,27 +51,33 @@ versions of Tor see [Section 3](#3-automatically-listen-on-tor).*
 The directory can be different of course, but (both) port numbers should be equal to
 your bitcoind's P2P listen port (8333 by default).
 
-	-externalip=X   You can tell bitcoin about its publicly reachable address using
-	                this option, and this can be a .onion address. Given the above
-	                configuration, you can find your .onion address in
-	                /var/lib/tor/bitcoin-service/hostname. For connections
-	                coming from unroutable addresses (such as 127.0.0.1, where the
-	                Tor proxy typically runs), .onion addresses are given
-	                preference for your node to advertise itself with.
-
-	-listen         You'll need to enable listening for incoming connections, as this
-	                is off by default behind a proxy.
-
-	-discover       When -externalip is specified, no attempt is made to discover local
-	                IPv4 or IPv6 addresses. If you want to run a dual stack, reachable
-	                from both Tor and IPv4 (or IPv6), you'll need to either pass your
-	                other addresses using -externalip, or explicitly enable -discover.
-	                Note that both addresses of a dual-stack system may be easily
-	                linkable using traffic analysis.
+    -externalip=X   You can tell bitcoin about its publicly reachable addresses
+                    using this option, and this can be a .onion address. Given
+                    the above configuration, you can find your .onion address in
+                    /var/lib/tor/bitcoin-service/hostname. For connections
```

Comment:
When a hidden service is created via tor-control, `/var/lib/tor/bitcoin-service/hostname` is not created (credits to @vasild).

Reply:
But this new text reads:

> Given the above configuration ...

and just a few lines above it says how to configure the hidden service via `torrc`:

> HiddenServiceDir /var/lib/tor/bitcoin-service/
> HiddenServicePort 8333 127.0.0.1:8333
> HiddenServicePort 18333 127.0.0.1:18333

So it is ok to mention `/var/lib/tor/bitcoin-service/hostname`.

I would suggest to drop the last line `HiddenServicePort 18333 127.0.0.1:18333` because it is for testnet and also to add `HiddenServiceVersion 3` so that the tor daemon creates a v3 service:

```
HiddenServiceDir /var/lib/tor/bitcoin-service/
HiddenServiceVersion 3
HiddenServicePort 8333 127.0.0.1:8333
```

Reply:
Does `bitcoind` configure its onion service via `torrc`?

Reply:
https://github.com/bitcoin/bitcoin/pull/19961#discussion_r492980674
Even given the above configuration, you make a point to consider we are talking about the operation of the parameter.


If the hidden service is created via tor-control and you need to discover it please check you debug.log for lines similar to the following, it should automatically be added as AddLocal
```
2020-08-11T13:42:26Z tor: Got service ID {random_service}, advertising service {random_service}.onion:8333
2020-08-11T13:42:26Z AddLocal({random_service}.onion:8333,4)
```